onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -253,7 +253,39 @@ void InputVariadicSelector::UpdateBuilder(NodesToOptimizeIndicesBuilder& builder
   builder.num_input_defs = 1;  // set to 1 as the first input is variadic
 }
 
-void OutputVariadicSelector::UpdateBuilder(NodesToOptimizeIndicesBuilder& builder) const {
+bool SplitNodeGroupSelector::Check(const GraphViewer& graph_viewer,
+                                   const Node& node,
+                                   const std::vector<const Node*>& dq_nodes,
+                                   const std::vector<const Node*>& q_nodes) const {
+  if (!CheckQDQNodes(graph_viewer, node, dq_nodes, q_nodes, 1)) {
+    return false;
+  }
+
+  auto get_const_initializer = [&graph_viewer](const std::string& initializer_name) {
+    return graph_viewer.GetConstantInitializer(initializer_name, true);
+  };
+
+  const Node& dq_node = *dq_nodes.front();
+  int32_t dt_input = dq_node.InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+
+  // All Q outputs should have same data type and (optionally) equal quantization parameters as the input.
+  for (size_t q_idx = 0; q_idx < q_nodes.size(); q_idx++) {
+    const Node& q_node = *q_nodes[q_idx];
+
+    if (dt_input != q_node.OutputDefs()[0]->TypeAsProto()->tensor_type().elem_type()) {
+      return false;
+    }
+
+    if (req_equal_quant_params_ &&
+        !IsQDQPairSupported(q_node, dq_node, get_const_initializer, graph_viewer.ModelPath())) {
+      return false;
+    }
+  }
```

Comment:
Out of interest, is it worth considering dropping the DQ/Q on the matching ones and copying the DQ to be prior to the Q on the non-matching, preserving the Q there as well? Not sure if there's any signficant perf gain possible.

e.g.  say Q2 does not match
```
DQin -> Split -> Q0 -> DQa -> S0
             \-> Q1 -> DQb -> S1
             \-> Q2 -> DQc -> S2
```

So we copy the DQ onto any non-matching path 
```
Split -> DQa -> S0
     \-> DQb -> S1
     \-> DQin -> Q2 -> DQc -> S2
```

Reply:
That seems worth investigating. Instead of just refusing to drop any DQ/Q ops, we can at least drop the ones that match. That said, I'm not in a position to be able to implement this transformation in this PR right now.